commands/http/handler.go

```diff
@@ -347,12 +346,17 @@ func allowOrigin(r *http.Request, cfg *ServerConfig) bool {
 		return true
 	}
 
+	u, err := url.Parse(origin)
+	if err != nil {
+		log.Error("incorrectly formatted origin on request: ", origin)
+	}
+
 	for _, o := range cfg.CORSOpts.AllowedOrigins {
 		if o == "*" { // ok! you asked for it!
 			return true
 		}
 
-		if o == origin { // allowed explicitly
```

Comment:
no, i think the protocol matters to CORS, one should be able to set CORS on https only :-1: